### PR TITLE
Update to golang/1.12.0

### PR DESCRIPTION
**Pull request by Liamsi**
- update docker image on circleci
 - remove GOCACHE=off from Makefile (see: https://tip.golang.org/doc/go1.12#gocache)

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by melekes** [comment]
Thanks 👍 Can you also update `https://img.shields.io/badge/go-1.10.4-blue.svg` in readme, scripts/install/*.sh files and Vagrantfile?

**Comment by Liamsi** [comment]
Thanks! Will do now.

**Comment by Liamsi** [comment]
There is also https://github.com/tendermint/tendermint/blob/e0f8936455029a40287a69d5b0e7baa4d5864da1/tools/build/Makefile#L7
which even contains references to deprecated (now private) repos: https://github.com/tendermint/tendermint/blob/e0f8936455029a40287a69d5b0e7baa4d5864da1/tools/build/Makefile#L79-L81

**Comment by Liamsi** [comment]
And https://github.com/tendermint/tendermint/blob/e0f8936455029a40287a69d5b0e7baa4d5864da1/networks/remote/integration.sh#L13-L14

File: README.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ For examples of the kinds of bugs we're looking for, see [SECURITY.md](SECURITY.
 
 Requirement|Notes
 ---|---
-Go version | Go1.11.4 or higher
```
Comment (melekes) [inline README.md]: Do we require 1.12 though? I thought running with 1.11.4 is still ok

Comment (Liamsi) [reply README.md]: You are right. 1.11.4 is still OK. It's just we only test with 1.12 from this PR on.

Comment (melekes) [reply README.md]: The fact that we only test 1.12 should not force us to require 1.12.

Comment (Liamsi) [reply README.md]: OK, you are right. I can revert that change. Shouldn't we then still ensure that all tests etc. run on all versions we support (looks like currently we could run on 1.10 - 1.12 without any changes but we are only sure about the latest version we test on).

Comment (melekes) [reply README.md]: We do need to test `release-X` branches that they compile with all versions we support. But that's not a priority and can be done separately.

**Pull request merged**